src/session.rs

```diff
@@ -0,0 +1,83 @@
+extern crate dirs;
+
+use crate::auth::Auth;
+use crate::request_items::Parameter;
+use crate::utils::ensure_session_dir_exists;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::prelude::*;
+
+const VERSION: &'static str = env!("CARGO_PKG_VERSION");
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Session {
+    pub identifier: String,
+    pub host: String,
+    pub headers: Vec<Parameter>,
+    pub auth: Option<Auth>,
+    ht_version: String,
+}
+
+impl Session {
+    pub fn new(identifier: String, host: &str, headers: Vec<Parameter>, auth: Option<Auth>) -> Session {
+        Session {
+            identifier,
+            host: host.to_string(),
+            headers,
+            auth,
+            ht_version: String::from(VERSION),
+        }
+    }
+
+    pub fn save(&self) -> std::io::Result<String> {
+        let json = serde_json::to_string(&self)?;
+        let identifier = &self.identifier;
```

Comment:
You can have values in a `String` that are not permitted by some filesystems.
Could this go through some check/conversion first?

Reply:
Ok, I think this is a valid concern. I tried to find a suitable validator for filenames, but most wouldn't fail something like `--session "http://microsoft.com/"`, as I think they should, to prevent the creation of a file with an error-prone filename. (note : httpie's `http  httpbin.org/basic-auth/me/pass --auth-type basic --auth me:pass --session "http://httpbin.org/" --verbose` doesn't fail either).
So I... gave up, and went for an easier path 😄 
I changed the sessions filenames to be 10-chars hashes (at the price of a bit more complexity and another dependency, on the blake2 crate, as it seems the most popular). So no more directly user-defined filenames, and as the identifier is still included as a json field, a user can still find which file contains which session.
What do you think ?

Reply:
That sounds great to me!

The owner/maintainer (not me) will need to approve the new dependency of course.

Reply:
Hmm. HTTPie checks for a path separator in the session name. If there is one, then it points directly to a file, so it's not relative to `~/.config/httpie/sessions/`.

It accepts `http://httpbin.org/` because it helpfully creates a directory called `http:` (in your current directory) to put the session file in.

But as long as we only ever put sessions in `~/.config` we don't have to handle that. So maybe this logic?
- If there's a path separator in the session name (`name.contains(std::path::is_separator)`), reject with a message saying explicit session filenames are not supported.
- Otherwise, just try creating the file, and if there's something else wrong with the filename you'll get an error to show to the user.

Reply:
Ok, it's simpler, and it fits well with the validation already in place, thanks.